packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -1,124 +1,114 @@
+      setDateType('pastDate');
+    } else if (dayjs(date).isAfter(dayjs(), 'date')) {
+      setDateType('futureDate');
+    } else {
+      setDateType('today');
+    }
+  }, [date]);
 
-  const filteredAppointments = appointmentServiceType
-    ? appointmentList.filter(({ serviceTypeUuid }) => serviceTypeUuid === appointmentServiceType)
-    : appointmentList;
-  const rowData = appointmentList.map((appointment, index) => {
-    return {
-      id: `${index}`,
-      ...appointment,
-    };
-  });
-  const filteredRow = appointmentServiceType
-    ? rowData.filter((app) => app.serviceTypeUuid === appointmentServiceType)
-    : rowData;
+  // set the default tab on first render
+  useEffect(() => {
+    if (!currentTab) {
+      setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+    }
+  }, [scheduledAppointmentPanels]);
 
-  const appointmentsBaseTableConfig = {
-    Scheduled: {
-      appointments: filteredAppointments,
-      isLoading,
-      tableHeading: t('scheduled', 'Scheduled'),
-      visits,
-      scheduleType,
-    },
-    CameEarly: {
-      appointments: earlyAppointmentList,
-      isLoading: loading,
-      tableHeading: t('cameEarly', 'Came early'),
-      visits,
-      scheduleType,
-    },
-    Honoured: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: t('honored', 'Honored'),
-      visits,
-      scheduleType,
-    },
-    Rescheduled: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: t('rescheduled', 'Rescheduled'),
-      visits,
-      scheduleType,
-    },
-    Pending: {
-      appointments: filteredRow,
-      isLoading,
-      tableHeading: isDateInPast ? t('missed', 'Missed') : t('notArrived', 'Not arrived'),
-      visits,
-      scheduleType,
-    },
-  };
+  // set the default tab whenever the date type changes
+  useEffect(() => {
+    setCurrentTab(getDefaultTab(scheduledAppointmentPanels));
+  }, [dateType]);
 
-  const currentConfig = appointmentsBaseTableConfig[scheduleType];
+  const getDefaultTab = (panels) => {
+    return panels?.find(
+      (panel) =>
+        (isDateInPast && panel.config.showForPastDate) ||
+        (isToday && panel.config.showForToday) ||
+        (isDateInFuture && panel.config.showForFutureDate),
+    )?.config.title;
+  };
 
   return (
     <>
-      {isToday && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setScheduleType(name)}>
-          <Switch name={'Scheduled'} text={t('scheduled', 'Scheduled')} />
-          <Switch name={'Honoured'} text={t('honored', 'Honored')} />
-          <Switch name={'Pending'} text={t('notArrived', 'Not arrived')} />
-          <Switch name={'CameEarly'} text={t('cameEarly', 'Came early')} />
+      {dateType === 'today' && (
+        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+          {scheduledAppointmentPanels
+            .filter((panel) => panel.config.showForToday)
```

Comment:
I would expect the "default" parameters for the extension to be ability here, but the config is null.  Do I need to look up the config somehow.
I also confirmed that if I *don't* register an extension config, the config is still null... while the docs suggest it should be getting the full module config in this case?

Reply:
`config` isn't a property of the extension like that. Can you point me to the docs so I can fix it? This is why we normally have `useConfig()` to get the actual configuration. It looks like there's also an API-gap here if we need to use it like this (so outside of the extension itself).

Reply:
There was no documentation, but "config" was a property of an extension, so I figured it might work and I couldn't find any documentation of how it should work?

Is there any example of how to use useConfig with an extension?

Reply:
You know, I just didn't realise that was a property! Learn something new all the time.

Usually `useConfig()` is done inside the component rendered in the extension; I don't think there's a way to use it outside of that context, since it relies on a context provider to actually calculate the configuration.

Reply:
@mogoodrich Is there only the one extension available? I realised this needed a slightly different approach so I implemented [basically this strategy](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1286#discussion_r1275392450).

![Screenshot of Expected appointments tab](https://github.com/openmrs/openmrs-esm-patient-management/assets/52504170/2809a673-8d69-41e1-84fe-e624d6586f82)

The issue here is that the documentation [here](https://github.com/openmrs/openmrs-esm-core/blob/3fb6a3ac0ce437148f59da083ee266d34fae5e53/packages/framework/esm-extensions/src/store.ts#L94) is inaccurate. Specifically it says that the config will be null "until the slot is mounted", when it should say "until the _extension is rendered_", which is quite different.

Reply:
At a technical level, what seems to happen is this:

1. The `Extension` object gets rendered
2. As part of that, [it calls the internal `renderExtension()` function](https://github.com/openmrs/openmrs-esm-core/blob/3fb6a3ac0ce437148f59da083ee266d34fae5e53/packages/framework/esm-react-utils/src/Extension.tsx#L79C15-L79C15), which [updates the internal extension store](https://github.com/openmrs/openmrs-esm-core/blob/3fb6a3ac0ce437148f59da083ee266d34fae5e53/packages/framework/esm-extensions/src/render.ts#L40C8-L40C8) add the new instance for the extension.
3. Once the internal extension store is updated, [the `updateConfigExtensionStore()` internal function is called](https://github.com/openmrs/openmrs-esm-core/blob/3fb6a3ac0ce437148f59da083ee266d34fae5e53/packages/framework/esm-extensions/src/store.ts#L147). This updates the `configExtensionStore` with the records of all extension _instances_.
4. The update to the `configExtensionStore` fires off [the `computeExtensionConfigs()` handler](https://github.com/openmrs/openmrs-esm-core/blob/3fb6a3ac0ce437148f59da083ee266d34fae5e53/packages/framework/esm-config/src/module-config/module-config.ts#L152) which actually computes each extensions configuration for each mounted instance (since different instances of extensions can have different configurations).

The up-shot of this is that an extension's `config` is `null` until the extension has been mounted and an additional tick or two has been run. And, thinking about it, I don't know that there's anyway to do the logic above in an improved way, since each mounted extension has basically a separate config.